integration/conftest.py

```diff
@@ -101,10 +108,48 @@ def cleanup():
     return tmp
 
 
+# NB: conceptually, it kind of makes sense to parametrize this fixture
+# on "Tahoe version" .. a quick attempt at that lead to all but the
+# first version failing; I think something doesn't get cleaned-up
+# properly between runs "or something". Instead, we use a command-line
+# option to specify tahoe-version and have multiple separate runs
+# (which looks better in CI anyway).
+@pytest.fixture(
+    scope='session',
+)
+def tahoe_venv(request, reactor, temp_dir):
```

Comment:
This should use https://docs.pytest.org/en/latest/reference/reference.html#std-fixture-tmp_path_factory instead. Requesting `tmpdir` (or `tmp_path`) forces that fixture to be session scoped as-well, so every test would be using the same temporary directory.

I think this is probably why trying to run multiple versions in the same process was failing.

Reply:
okay .. that could be interesting to fix (separately) if that's indeed the problem. (It almost seems better to have a command-line option anyway, so that we see 3x top-level CI runs instead of one with 3x tahoe versions "inside" it ...)

Reply:
Yeah, I think having the command line option is reasonable, I'm not suggesting removing it. We definitely want to fix this fixture, so it isn't turning the function-scoped fixture `temp_dir` into a session-scoped one.